source/common/upstream/upstream_impl.cc

```diff
@@ -1127,10 +1127,20 @@ void StrictDnsClusterImpl::updateAllHosts(const HostVector& hosts_added,
   // At this point we know that we are different so make a new host list and notify.
   for (const ResolveTargetPtr& target : resolve_targets_) {
     priority_state_manager.initializePriorityFor(target->locality_lb_endpoint_);
-    for (const HostSharedPtr& host : target->hosts_) {
-      if (target->locality_lb_endpoint_.priority() == current_priority) {
-        priority_state_manager.registerHostForPriority(host, target->locality_lb_endpoint_,
-                                                       target->lb_endpoint_, absl::nullopt);
+    std::unordered_set<std::string> host_addresses;
```

Comment:
Sure will add more comments here. While I think this dedupe here probably is not required anymore since the underlying problem is indeed when we update the internal `all_hosts_` data (for EDS (STATIC), the updated hosts consist of all host sets, while for STRICT_DNS it is not, depends on each DNS resolution).

Will make sure the priority state manager can actually handle this (no duplications).

Reply:
@mattklein123 for following up this issue, I have opened one here: https://github.com/envoyproxy/envoy/issues/4590 to track. This is based on @snowp's comment. I'll take a look at that later.